motionplan/solverFrame.go

```diff
@@ -200,6 +200,32 @@ func (sf *solverFrame) Transform(inputs []frame.Input) (spatial.Pose, error) {
 	return tf.(*frame.PoseInFrame).Pose(), nil
 }
 
+// Interpolate interpolates the given amount between the two sets of inputs.
+func (sf *solverFrame) Interpolate(from, to []frame.Input, by float64) ([]frame.Input, error) {
+	if len(from) != len(sf.DoF()) {
+		return nil, frame.NewIncorrectInputLengthError(len(from), len(sf.DoF()))
+	}
+	if len(to) != len(sf.DoF()) {
+		return nil, frame.NewIncorrectInputLengthError(len(to), len(sf.DoF()))
+	}
+	interp := make([]frame.Input, 0, len(to))
+	posIdx := 0
+	for _, transform := range sf.frames {
```

Comment:
[q]: why do we call this "transform", I think we could use a slight better name here.

Reply:
agree.  Probably clearer to use `frame` so that we utilize the idiom of `for _, object := range objects`

Reply:
Good call. This code dates back to early 2021 before we had a super strong concept of "frame" beyond just "spatial transform", now's a good time to update the nomenclature.